### PR TITLE
Adds wheel

### PR DESCRIPTION
**Pull request by amdcaruso**
This is a spinning wheel made with vanilla js + css to be used during standup.
Its aim is to be used for mentoring a lady learn the foundation of js + css with a practical example

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -10,18 +10,18 @@
...
-            <span class="person person2"><img class="selected" src="./assets/images/sticker.png" />80</span>
-            <span class="person person3"><img class="selected" src="./assets/images/sticker.png" />125</span>
-            <span class="person person4"><img class="selected" src="./assets/images/sticker.png" />170</span>
-            <span class="person person5"><img class="selected" src="./assets/images/sticker.png" />215</span>
-            <span class="person person6"><img class="selected" src="./assets/images/sticker.png" />260</span>
-            <span class="person person7"><img class="selected" src="./assets/images/sticker.png" />305</span>
-            <span class="person person8"><img class="selected" src="./assets/images/sticker.png" />340</span>
+            <span class="person person0">Stef</span>
+            <span class="person person1">Jeanie</span>
+            <span class="person person2">Niro</span>
```
Comment (inPhoenix) [inline index.html:19]: Could it be an array of names? Something like:

const names = ["stef", "Jeanie" ...]

names.map( (name, n) => {
<span class=`person person${n}`> name  <span>
} )

File: script.js [hunk 2]
```diff
@@ -3,19 +3,33 @@
     const box = document.querySelector('.box');
     const spin = document.querySelector('.spin');
     const marker = document.querySelector('.marker');
-    const person = document.querySelector('.person');
-    const selected = document.querySelector('.selected');
 
     // Defines variables
     let deg = 0;
-    let teamMembers = 8;
     let currentWinningNumber = 0;
```
Comment (inPhoenix) [inline script.js:9]: use const for not modified variables

File: script.js [hunk 3]
```diff
@@ -3,19 +3,33 @@
...
+    // const person0 = document.querySelector('#box > span.person.person0');
+    // const person1 = document.querySelector('#box > span.person.person1');
+    // const person2 = document.querySelector('#box > span.person.person2');
+    // const person3 = document.querySelector('#box > span.person.person3');
+    // const person4 = document.querySelector('#box > span.person.person4');
+    // const person5 = document.querySelector('#box > span.person.person5');
+    // const person6 = document.querySelector('#box > span.person.person6');
+    // const person7 = document.querySelector('#box > span.person.person7');
+
+    for (let i = 0; i < teamMembers; i++) {
```
Comment (inPhoenix) [inline script.js:29]: teamMembers.forEach((item, number ) => {

})

File: script.js [hunk 4]
```diff
@@ -52,53 +66,71 @@
         }
     })();
 
+    // namesEachPerson
+    (function nameEachPerson() {
```
Comment (inPhoenix) [inline script.js:70]: I would not use an auto invoked function. Do it as a named function and invoked later.

File: script.js [hunk 5]
```diff
@@ -52,53 +66,71 @@
...
         );
-        winningOrder.splice(currentWinningNumber, 1);
+        segmentArray.splice(currentWinningNumber, 1);
         return landedNumber;
     }
 
     // Checks if everyone has been selected
     function hasEveryOneBeenSelected() {
-        if (winningOrder === undefined || winningOrder.length === 0) {
+        if (segmentArray === undefined || segmentArray.length === 0) {
```
Comment (inPhoenix) [inline script.js:103]: This if could be !segmentArray

File: script.js [hunk 6]
```diff
@@ -52,53 +66,71 @@
...
-            winningOrder,
+            segmentArray,
             currentWinningNumber,
-            winningOrder[currentWinningNumber],
-            segment[winningOrder[currentWinningNumber]]
+            segmentArray[currentWinningNumber],
+            segment[segmentArray[currentWinningNumber]]
         );
-        winningOrder.splice(currentWinningNumber, 1);
+        segmentArray.splice(currentWinningNumber, 1);
```
Comment (inPhoenix) [inline script.js:97]: splice changes the original array. Its ok but slice is likely to be more safe.
www.educative.io/edpresso/splice-vs-slice-in-javascript?utm_source=Google%20AdWords&aid=5082902844932096&utm_medium=cpc&utm_campaign=kb-dynamic-edpresso&gclid=Cj0KCQiAqo3-BRDoARIsAE5vnaIVf8yihL1DBnDm1wcZ8aKbR7QTigcGA6FSQEtvvCCxVxBcxj1DQpMaAoX4EALw_wcB

File: script.js [hunk 7]
```diff
@@ -52,53 +66,71 @@
...
-    // Function to make the selected sticker visible
-    function showSelected() {
-        if (selected.style.display === 'none') {
-            selected.style.display = 'block';
-        }
-    }
+    // Makes the selected sticker visible
+    function addsSticker(person) {
+        let url = './assets/images/sticker.png';
+        let image = new Image();
```
Comment (inPhoenix) [inline script.js:113]: use const if it will not be reassigned.

File: script.js (hunk 5 above)
Comment (amdcaruso) [reply script.js:103]: :-( Unfortunately with the way the code is currently written if I do it simply as `if (!segmentArray) {` I get an error, the current way it's written is working. Thanks for the review!

File: index.html (hunk 1 above)
Comment (amdcaruso) [reply index.html:19]: That would be a great idea but the names are placeholders for now, they'll be images with the faces of those names!

File: script.js [hunk 8]
```diff
@@ -3,19 +3,25 @@
...
 
     // Defines variables
     let deg = 0;
-    let teamMembers = 8;
-    let currentWinningNumber = 0;
-    let segment = new Array(teamMembers);
-    let winningOrder = [40, 80, 125, 170, 215, 260, 305, 340];
+    let currentSelectedIndex = 0;
+    const numberOfTeamMembers = 8;
+    let segment = new Array(numberOfTeamMembers);
```
Comment (GustavoCostaW) [inline script.js:11]: I would use the literal notation for being faster as well.

ref: https://stackoverflow.com/a/3500144/1417551

File: script.js [hunk 9]
```diff
@@ -3,19 +3,25 @@
...
+    const personArray = new Array(numberOfTeamMembers);
```
Comment (GustavoCostaW) [inline script.js:12]: I would use the literal notation for being faster as well.

ref: https://stackoverflow.com/a/3500144/1417551

File: script.js [hunk 10]
```diff
@@ -3,19 +3,25 @@
...
+    const segmentArray = [40, 80, 125, 170, 215, 260, 305, 340];
+    let segmentArrayPerson = new Array(numberOfTeamMembers);
```
Comment (GustavoCostaW) [inline script.js:14]: I would use the literal notation for being faster as well.

ref: https://stackoverflow.com/a/3500144/1417551

File: script.js [hunk 11]
```diff
@@ -52,53 +58,71 @@
...
-        showSelected();
-    });
+        let toBeAppended = segmentArrayPerson[person];
+        toBeAppended.appendChild(image);
+    }
 
-    // On click on spin
+    // On click on the spin button
     spin.addEventListener('click', () => {
         hasEveryOneBeenSelected();
```
Comment (GustavoCostaW) [inline script.js:115]: why don't you make it returns a boolean? 

e.g:

```javascript
    function hasEveryOneBeenSelected() {
        if (segmentArray === undefined || segmentArray.length === 0) {
            alert("Everyone's been selected!");
            document.getElementById('spin').disabled = true;
            return true;
        }
    return false;
    }



spin.addEventListener('click', () => {
   if (!hasEveryOneBeenSelected()) {
      // make the animation...
   }
}
```

**Pull request merged**